ext/pgsql/pgsql.c

```diff
@@ -115,15 +134,135 @@ ZEND_TSRMLS_CACHE_DEFINE()
+	zend_throw_error(NULL, "Cannot directly construct PgSql, use pg_connect() or pg_pconnect() instead");
+	return NULL;
+}
+
+static void pgsql_link_free(pgsql_link_handle *link)
+{
+	PGresult *res;
+	zval *hash;
+
+	while ((res = PQgetResult(link->conn))) {
+		PQclear(res);
+	}
+	PQfinish(link->conn);
+	PGG(num_links)--;
+
+	/* Remove connection hash for this link */
+	hash = zend_hash_index_find(&PGG(hashes), (uintptr_t) link->conn);
+	if (hash) {
+		zend_hash_index_del(&PGG(hashes), (uintptr_t) link->conn);
+		zend_hash_del(&PGG(regular_list), Z_STR_P(hash));
+	}
+
+	link->conn = NULL;
+}
+
+static void pgsql_link_free_obj(zend_object *obj)
+{
+	pgsql_link_handle *link = pgsql_link_from_obj(obj);
+
+	if (link->conn) {
+		pgsql_link_free(link);
+	}
+
+	zend_object_std_dtor(&link->std);
+}
+
+static inline pgsql_result_handle *pgsql_result_from_obj(zend_object *obj) {
+	return (pgsql_result_handle *)((char *)(obj) - XtOffsetOf(pgsql_result_handle, std));
+}
+
+#define Z_PGSQL_RESULT_P(zv) pgsql_result_from_obj(Z_OBJ_P(zv))
+
+static zend_object *pgsql_result_create_object(zend_class_entry *class_type) {
+	pgsql_result_handle *intern = zend_object_alloc(sizeof(pgsql_result_handle), class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+	intern->std.handlers = &pgsql_result_object_handlers;
+
+	return &intern->std;
+}
+
+static zend_function *pgsql_result_get_constructor(zend_object *object) {
+	zend_throw_error(NULL, "Cannot directly construct PgSqlResult, use a dedicated function instead");
+	return NULL;
+}
+
+static void pgsql_result_free(pgsql_result_handle *pg_result)
+{
+	PQclear(pg_result->result);
+}
+
+static void pgsql_result_free_obj(zend_object *obj)
+{
+	pgsql_result_handle *pg_result = pgsql_result_from_obj(obj);
+
+	if (pg_result->result) {
+		pgsql_result_free(pg_result);
+	}
+
+	zend_object_std_dtor(&pg_result->std);
+}
+
+static inline pgLofp *pgsql_lob_from_obj(zend_object *obj) {
+	return (pgLofp *)((char *)(obj) - XtOffsetOf(pgLofp, std));
+}
+
+#define Z_PGSQL_LOB_P(zv) pgsql_lob_from_obj(Z_OBJ_P(zv))
+
+static zend_object *pgsql_lob_create_object(zend_class_entry *class_type) {
+	pgLofp *intern = zend_object_alloc(sizeof(pgLofp), class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+	intern->std.handlers = &pgsql_lob_object_handlers;
+
+	return &intern->std;
+}
+
+static zend_function *pgsql_lob_get_constructor(zend_object *object) {
+	zend_throw_error(NULL, "Cannot directly construct PgSqlLob, use pg_lo_open() instead");
+	return NULL;
+}
+
+static void pgsql_lob_free_obj(zend_object *obj)
+{
+	pgLofp *lofp = pgsql_lob_from_obj(obj);
+
+	zend_object_std_dtor(&lofp->std);
+}
 
 /* Compatibility definitions */
 
 #ifndef HAVE_PGSQL_WITH_MULTIBYTE_SUPPORT
 #define pg_encoding_to_char(x) "SQL_ASCII"
 #endif
 
-/* {{{ _php_pgsql_trim_message */
```

Comment:
I removed some of these unnecessary comments because they annoyed me at some point

Reply:
This does make a lot of noise in the PR sadly :-/

Reply:
I agree that it was not a good idea to do this here. I can get rid of these changes If they make code review difficult

Reply:
I'll do it separately!